### PR TITLE
feat(ironfish): Update multisig store to use identity as a key

### PR DESCRIPTION
**Pull request by rohanjadvani**
## Summary

When trying to add authentication to signing commitments, there seems to be missing links between pieces of our data. Namely, there's no explicit link between what data is imported and any existing secrets. This PR attempts to glue all of that together. 

What this is trying to fix:
* Make it UX friendly by defaulting the name of an imported participant to the same as the secret name (before it was just an identifier)
* Make an abstraction around the signing commitment so it's easy to verify (will be in the subsequent PR)
* Make the multisig store easier to use / less error prone

Changes:
* Updates `ironfish-frost` dependency 
* Add Napi type for signed commitment with useful methods instead of deserializing in JS
* Update creating signature commitment and signature share to use the multisig secret
* Store the account secret instead of identity
* Updates the multisig secret store to be `identity` -> `{ secret, name }`. I don't think it's a good idea to have a `name` as a primary key, though I'm happy to change that if there's a strong argument.
* Update imports to accept `secret` as well as `identity`. If it's from TDK, require an existing secret.

## Testing Plan

* Update unit tests to save multisig secret before querying for corresponding identity
* Update import/export account test cases

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[x] Yes
```

There are breaking changes, but this doesn't require any special treatment because a lot of changes are still in staging and no one is using this feature yet.

### CODE REVIEW

File: ironfish/src/wallet/walletdb/accountValue.ts [hunk 1]
```diff
@@ -27,7 +27,18 @@ export interface AccountValue {
...
+  name: string
+  spendingKey: string | null
+  viewKey: string
+  incomingViewKey: string
+  outgoingViewKey: string
+  publicAddress: string
+  createdAt: HeadValue | null
+  multisigKeys?: MultisigKeysImport
+  proofAuthorizingKey: string | null
+}
```
Comment (patnir) [inline ironfish/src/wallet/walletdb/accountValue.ts:42]: why this change? Seems like `Omit<AccountValue, 'id'>` would be cleaner than repeating the fields in two places?

**Review by hughy — APPROVED** [review]

File: ironfish/src/wallet/walletdb/accountValue.ts (hunk 1 above)
Comment (rohanjadvani) [reply ironfish/src/wallet/walletdb/accountValue.ts:42]: 1. I don't really like `Omit` in this case because it's not clear that the import data could be slightly different.
2. The import data is slightly different. Multi sig keys can have either an identity or secret, but we store the secret.

File: ironfish-rust-nodejs/index.d.ts [hunk 2]
```diff
@@ -4,8 +4,9 @@
 /* auto-generated by NAPI-RS */
 
 export const IDENTITY_LEN: number
```
[the remark this answers is not in the document]
Comment (rohanjadvani) [reply ironfish-rust-nodejs/index.d.ts:6]: This is still used in participant identity encoding.

File: ironfish/src/rpc/routes/wallet/types.ts [hunk 3]
```diff
@@ -156,6 +156,7 @@ export const RpcWalletTransactionSchema: yup.ObjectSchema<RpcWalletTransaction>
 
 export type RpcMultisigKeys = {
   identity?: string
+  secret?: string
```
[the remark this answers is not in the document]
Comment (rohanjadvani) [reply ironfish/src/rpc/routes/wallet/types.ts:159]: I think we need both for TDK and regular account export.

File: ironfish/src/wallet/wallet.ts [hunk 4]
```diff
@@ -1518,10 +1519,33 @@ export class Wallet {
...
+      isMultisigSignerTrustedDealerImport(accountValue.multisigKeys)
+    ) {
+      const multisigSecret = await this.walletDb.getMultisigSecret(
+        Buffer.from(accountValue.multisigKeys.identity, 'hex'),
+      )
+      if (!multisigSecret) {
+        throw new Error('Cannot import identity without a corresponding multisig secret')
+      }
+
+      name = multisigSecret.name
```
[the remark this answers is not in the document]
Comment (rohanjadvani) [reply ironfish/src/wallet/wallet.ts:1537]: Will do in subsequent PR
